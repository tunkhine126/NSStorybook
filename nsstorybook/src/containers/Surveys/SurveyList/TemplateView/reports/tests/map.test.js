/**
 *
 * Tests for Map
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import TestingComponent from 'components/shared/Testing';
import { render } from '@testing-library/react';

import Map from '../map';

describe('<Map />', () => {
  const props = {
    responses: [],
    surveyName: 'Test Survey',
    primaryRecipientIdPath: '/recipients/family-list/family/:id',
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <Map {...props} />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <Map {...props} />
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
