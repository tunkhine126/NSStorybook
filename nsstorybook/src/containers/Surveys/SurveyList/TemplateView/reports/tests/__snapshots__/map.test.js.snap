// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map /> Should render and match the snapshot 1`] = `
<div
  style="position: relative; padding-left: 40px; padding-right: 40px; width: 100%; height: 400px; cursor: grab; outline: none;"
  tabindex="0"
>
  <div
    style="position: relative; width: 100%; height: 100%;"
  >
    <div
      class=""
      style="position: absolute; width: 100%; height: 100%; overflow: hidden; visibility: inherit;"
    />
    <div
      style="overflow: visible; height: 0px; width: 0px;"
    />
    <div
      class="resize-triggers"
    >
      <div
        class="expand-trigger"
      >
        <div
          style="width: 1px; height: 1px;"
        />
      </div>
      <div
        class="contract-trigger"
      />
    </div>
  </div>
</div>
`;
